### PR TITLE
LPS-157987 check permissions to user/guest to see the file

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
- LPS-157987 check expired FileEntry on the case of friendlyURL
- LPS-157987 check permissions to user/guest to see the file

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:stop

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
@adolfopa 
Once we check the permissions, 
This is what we saw access
before without friendlyURL :
image
after with friendlyURL :
image

Thats why I added the last 2 commits because I do not know how to check the permissions on the WebServerServelt:
https://github.com/liferay-lima/liferay-portal/pull/3238/commits/eb12ebbed788a215f5ba11f9f8ca9851f08d8f7f

https://github.com/liferay-lima/liferay-portal/pull/3238/commits/a825e575005d07563b5341cb1c2e43d7c84b5b4e


Any idea, welcome

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

File: portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java [hunk 1]
```diff
@@ -1680,10 +1680,14 @@ else if (_PATH_SEPARATOR_FILE_ENTRY.equals(pathArray[0])) {
 			Optional<FileEntry> fileEntryOptional = _resolveFileEntry(
 				httpServletRequest, pathArray);
 
-			return fileEntryOptional.orElseThrow(
+			FileEntry fileEntry = fileEntryOptional.orElseThrow(
 				() -> new NoSuchFileEntryException(
 					"No file entry found for friendly URL " +
 						Arrays.toString(pathArray)));
+
+			_checkExpiredFileEntry(fileEntry, httpServletRequest);
```
Comment (adolfopa) [inline portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java:1689]: Why not make the permission checks in the WebServerServlet as well? That way the lower layers can keep using the local service (that makes the useful in more contexts).

Comment (AliciaGarciaGarcia) [reply portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java:1689]: Because I wanted to mimic the behavior on the  DLAppServiceUtil.getFileEntryByUuidAndGroupId but I ended making the thing too complex.

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/friendly/url/resolver/FileEntryFriendlyURLResolverImpl.java [hunk 2]
```diff
@@ -35,7 +38,7 @@
 
 	@Override
 	public Optional<FileEntry> resolveFriendlyURL(
-			long groupId, String friendlyURL)
+			long groupId, boolean checkPermissions, String friendlyURL)
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/friendly/url/resolver/FileEntryFriendlyURLResolverImpl.java:41]: @AliciaGarciaGarcia I really dislike this. From my point of view, the correct approach would be:
1. Let the friendly url resolver work with local services. Its purpose is to resolve a friendly URL to a file entry, nothing else, nothing more. No permission checks, no expiration date checks. This will make it simpler and reusable in other contexts.
2. Make all the checks in the caller. In this case, the web server servlet should ensure that only non expired file entries for which the user has view permissions are visible.

Change the _checkExpiredFileEntry method so that it checks both the expiration date and the VIEW permission (you should rename it as well).

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

File: portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java [hunk 3]
```diff
@@ -1794,6 +1779,50 @@ private boolean _processCompanyInactiveRequest(
 		return true;
 	}
 
+	private void _verifyFileEntry(
```
Comment (adolfopa) [inline portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java:1782]: Can you rename this method to `_check*`? "Verify" has a very specific meaning that is not related to this usage.

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:document

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by brianchandotcom** [comment]
@adolfopa @AliciaGarciaGarcia merged to master
